### PR TITLE
build: reinstate xgo-driven darwin builds

### PR DESCRIPTION
**Pull request by tbg**
~~@jseldess tagging you since the S3 filename of new OSX binaries will change. I don't know if that affects you at all, so if it doesn't feel free to remove the label.~~ nevermind, retracted the version upgrade.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/6051)

### CODE REVIEW

File: build/push-tagged-aws.sh [hunk 1]
```diff
@@ -24,5 +24,4 @@ cd "$(dirname "${0}")"
 # Tar up and push linux/osx binaries.
 ./push-release-tarball.sh cockroach cockroach-${CIRCLE_TAG}.linux-amd64
 # The OSX binary name is generated by build-osx.sh.
-# TODO(tschottdorf): temporarily disabled, #6030.
-# ./push-release-tarball.sh cockroach-darwin-10.9-amd64 cockroach-${CIRCLE_TAG}.darwin-10.9-amd64
+./push-release-tarball.sh cockroach-darwin-10.11-amd64 cockroach-${CIRCLE_TAG}.darwin-10.9-amd64
```
Comment (bdarnell) [inline build/push-tagged-aws.sh]: The second occurrence of `10.9` looks like it needs to change too.

**Comment by bdarnell** [comment]
Does this mean that our binaries will no longer run on OSX 10.9 and 10.10? That's unfortunate. Is it not possible to maintain compatibility with older versions?

**Comment by tschottdorf** [comment]
I'm not sure about the move to .11, it's not a necessity. We can build
everything from 10.6 to 10.11, but I don't know what the compatibility
story is.

On Wed, Apr 13, 2016 at 4:48 PM Ben Darnell <notifications@github.com>
wrote:

> Does this mean that our binaries will no longer run on OSX 10.9 and 10.10?
> That's unfortunate. Is it not possible to maintain compatibility with older
> versions?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6051#issuecomment-209641541>
>
-- 

-- Tobias

**Comment by bdarnell** [comment]
I'd rather continue to target 10.9 unless there's a reason to change.

**Comment by tschottdorf** [comment]
Ok, building for 10.9. Tried locally, binary works fine as far as I can tell.

File: build/push-tagged-aws.sh (hunk 1 above)
Comment (tschottdorf) [inline build/push-tagged-aws.sh]: Moot with the corresponding commit removed.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**